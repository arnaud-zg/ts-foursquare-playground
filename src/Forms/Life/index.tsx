export { PutCredentialsForm } from './PutCredentialsForm.container'
export { GetLifeForm } from './GetLifeForm.container'
