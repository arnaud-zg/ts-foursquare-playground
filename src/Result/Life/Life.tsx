import React from 'react'
import { Props } from './index'

export class Life extends React.Component<Props> {
  render() {
    const { life, credentials } = this.props
    return (
      <div>
        <div>
          <h3>Life</h3>
          <code>
            <pre>{JSON.stringify(life)}</pre>
          </code>
        </div>
        <div className="mt-4">
          <h3>Credentials</h3>
          <code>
            <pre>{JSON.stringify(credentials)}</pre>
          </code>
        </div>
      </div>
    )
  }
}
