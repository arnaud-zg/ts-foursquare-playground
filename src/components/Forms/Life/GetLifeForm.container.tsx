import { connect } from 'react-redux'
import { getLife, initialState } from 'ts-foursquare'
import { GetLifeForm as GetLifeFormComponent } from './GetLifeForm'

interface ContainerProps {}

export type Props = ContainerProps &
  ReturnType<typeof mapStateToProps> &
  typeof mapDispatchToProps

const mapStateToProps = (state: typeof initialState) => ({
  life: state.life,
})

const mapDispatchToProps = {
  getLife,
}

export const GetLifeForm = connect(
  mapStateToProps,
  mapDispatchToProps
)(GetLifeFormComponent)
