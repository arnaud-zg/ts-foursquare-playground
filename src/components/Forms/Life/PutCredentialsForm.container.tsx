import { connect } from 'react-redux'
import {
  initialState,
  credentialsSelector,
  putCredentials,
} from 'ts-foursquare'
import { NResolve } from 'ts-foursquare/types/resolve'
import { PutCredentialsForm as PutCredentialsFormComponent } from './PutCredentialsForm'

interface ContainerProps {
  onSubmit?: (values: NResolve.ICredentialsPayload) => void
}

export type Props = ContainerProps &
  ReturnType<typeof mapStateToProps> &
  typeof mapDispatchToProps

const mapStateToProps = (state: typeof initialState) => ({
  credentials: credentialsSelector(state),
})

const mapDispatchToProps = {
  putCredentials,
}

export const PutCredentialsForm = connect(
  mapStateToProps,
  mapDispatchToProps
)(PutCredentialsFormComponent)
