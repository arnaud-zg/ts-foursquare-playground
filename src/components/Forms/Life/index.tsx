export { GetLifeForm } from './GetLifeForm'
export { PutCredentialsForm } from './PutCredentialsForm'
