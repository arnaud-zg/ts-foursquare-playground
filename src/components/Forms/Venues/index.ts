export { GetVenuesByQueryForm } from './GetVenuesByQueryForm'
export { GetVenuesByPlaceGeocode } from './GetVenuesByPlaceForm'
