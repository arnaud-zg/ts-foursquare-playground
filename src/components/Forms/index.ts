export * from './Life'
export * from './Venues'
