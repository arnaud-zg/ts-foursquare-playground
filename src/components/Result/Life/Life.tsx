import React from 'react'
import ReactJsonView from 'react-json-view'
import { Props } from './index'

export class Life extends React.Component<Props> {
  render() {
    const { life, credentials } = this.props
    return (
      <div>
        <div>
          <h3>Life</h3>
          <ReactJsonView src={life} />
        </div>
        <div className="mt-4">
          <h3>Credentials</h3>
          <ReactJsonView src={credentials} />
        </div>
      </div>
    )
  }
}
