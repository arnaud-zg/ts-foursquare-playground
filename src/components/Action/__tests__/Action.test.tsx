import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import {
  configureStore,
  getVenuesSearchAsync,
  putCredentials,
} from 'ts-foursquare'
import { Action } from '..'
import { PutCredentialsForm } from '../../Forms/Life'
import { GetVenuesSearchForm } from '../../Forms'

const store = configureStore({ middlewares: [] })

describe('Component/Action', () => {
  it('should make a snapshot after action: ELifeAction.PUT_CREDENTIALS', () => {
    const props = {
      actionCreator: putCredentials,
      actionPayload: { clientId: '', clientSecret: '' },
      renderFormPayload: () => <PutCredentialsForm />,
    }
    const tree = renderer
      .create(
        <Provider store={store}>
          <Action {...props} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should make a snapshot after action: EVenuesAction.GET_VENUES_SEARCH_REQUEST', () => {
    const initialValues = { query: '' }
    const props = {
      actionCreator: getVenuesSearchAsync.request,
      actionPayload: initialValues,
      renderFormPayload: () => (
        <GetVenuesSearchForm initialValues={initialValues} />
      ),
    }
    const tree = renderer
      .create(
        <Provider store={store}>
          <Action {...props} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
