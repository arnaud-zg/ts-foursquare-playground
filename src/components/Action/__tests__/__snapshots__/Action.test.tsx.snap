// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/Action renders correctly 1`] = `
<section
  className="shadow"
>
  <article
    className="flex-1 border-b"
  >
    <div
      className="h-full border-l-2 bg-grey-lightest border-indigo"
    >
      <header
        className="flex justify-between items-center p-5 pl-8 pr-8 cursor-pointer select-none"
      >
        <span
          className="text-indigo font-thin text-xl"
        >
          RESOLVE_PUT_CREDENTIALS | type: object
        </span>
      </header>
      <div
        className="pl-8 pr-8 pb-5 text-grey-darkest select-none"
      >
        <div
          className="flex mt-2"
        >
          <div
            className="flex-1 self-center"
          >
            <div
              className="flex"
            >
              <h2>
                Action
              </h2>
              <pre
                className="ml-2"
              >
                <code>
                  {"type":"RESOLVE_PUT_CREDENTIALS","payload":{"clientId":"","clientSecret":""}}
                </code>
              </pre>
            </div>
            <div
              className="flex"
            >
              <h2>
                Payload
              </h2>
              <p
                className="ml-2"
              >
                <pre
                  className="ml-2"
                >
                  <code>
                    {"clientId":"","clientSecret":""}
                  </code>
                </pre>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
  <div
    className="flex"
  >
    <article
      className="flex-1 border-b"
    >
      <div
        className="h-full border-l-2 bg-grey-lightest border-indigo"
      >
        <header
          className="flex justify-between items-center p-5 pl-8 pr-8 cursor-pointer select-none"
        >
          <span
            className="text-indigo font-thin text-xl"
          >
            Payload
          </span>
        </header>
        <div
          className="pl-8 pr-8 pb-5 text-grey-darkest select-none"
        >
          <form
            className="max-w-xl m-2"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <p
              className="mt-4 text-gray-800 font-medium"
            >
              Credentials
            </p>
            <div>
              <label
                className="block text-sm text-gray-600"
                htmlFor="clientId"
              >
                clientId
              </label>
              <input
                className="w-full px-5 py-2 text-gray-700 bg-gray-200 rounded"
                name="clientId"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Ex: 1234"
                type="text"
                value=""
              />
            </div>
            <div
              className="mt-4"
            >
              <label
                className="block text-sm text-gray-600"
                htmlFor="clientSecret"
              >
                clientSecret
              </label>
              <input
                className="w-full px-5 py-2 text-gray-700 bg-gray-200 rounded"
                name="clientSecret"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Ex: 1234"
                type="text"
                value=""
              />
            </div>
            <div
              className="mt-4"
            >
              <div
                className="flex justify-end"
              >
                <button
                  className="bg-white text-gray-800 font-bold rounded border-b-2 border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white shadow-md py-2 px-6 inline-flex items-center"
                  type="submit"
                >
                  <span
                    className="mr-2"
                  >
                    Send
                  </span>
                  <svg
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                      fill="currentcolor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </article>
    <article
      className="flex-1 border-b"
    >
      <div
        className="h-full border-l-2 bg-grey-lightest border-indigo"
      >
        <header
          className="flex justify-between items-center p-5 pl-8 pr-8 cursor-pointer select-none"
        >
          <span
            className="text-indigo font-thin text-xl"
          >
            Result
          </span>
        </header>
        <div
          className="pl-8 pr-8 pb-5 text-grey-darkest select-none"
        >
          <div>
            <div>
              <h3>
                Life
              </h3>
              <code>
                <pre>
                  {"life":false,"credentials":{"clientId":"","clientSecret":""}}
                </pre>
              </code>
            </div>
            <div
              className="mt-4"
            >
              <h3>
                Credentials
              </h3>
              <code>
                <pre>
                  {"clientId":"","clientSecret":""}
                </pre>
              </code>
            </div>
          </div>
        </div>
      </div>
    </article>
  </div>
</section>
`;
